### ISSUE TITLE
[BUG] New transformers updates have broken the HF and OpenClip integrations in FiftyOne

### ISSUE THREAD

**Issue by danielgural** [voxel51/fiftyone#5657]
### Describe the problem

New `transformers` updates have broken the HF and OpenClip integrations in FiftyOne

### Code to reproduce issue

Many issues, will update thread with known related issues, more investigation needed.

### System information

-   **OS Platform and Distribution** Mac
-   **Python version** (`python --version`): 3.10
-   **FiftyOne version** (`fiftyone --version`): develop
-   **FiftyOne installed from** (pip or source): source

**Comment by danielgural** [comment]
All on transfomers 4.Bug with zero-shot classification:

```
import fiftyone as fo
import fiftyone.zoo as foz

dataset = foz.load_zoo_dataset("quickstart")  


siglip_model = foz.load_zoo_model(
    name_or_url="zero-shot-classification-transformer-torch",
    name_or_path="google/siglip2-so400m-patch14-384",
    classes=dataset.distinct("ground_truth.detections.label"),
    # install_requirements=True # uncomment this line if you are running this code for the first time
    )

dataset.apply_model(siglip_model, label_field="class_predictions")
```
outputs garbage

Similar model: 
```
model = foz.load_zoo_model(
    "zero-shot-classification-transformer-torch",
    classes=dataset.distinct("ground_truth.detections.label"),
    name_or_path="google/siglip2-base-patch16-224",  # HF model name or path
)

dataset.apply_model(model, label_field="classif_predictions")
```
outputs garbage

Open Clip:
```
siglip = foz.load_zoo_model(
    "open-clip-torch",
    clip_model="hf-hub:timm/ViT-B-16-SigLIP",
    pretrained="",
    text_prompt="A photo of a",
    classes=["person", "dog", "cat", "bird", "car", "tree", "chair"],
)
dataset.apply_model(siglip, label_field="clip_predictions")
```
Works 

Alt clip via HF integration

```
model = foz.load_zoo_model(
    "zero-shot-classification-transformer-torch",
    name_or_path="BAAI/AltCLIP",  # HF model name or path
    classes=["cat", "dog", "bird", "fish", "turtle"],  # optional
)
dataset.apply_model(model, label_field="altclip_predictions")
```

Fine
```
model = foz.load_zoo_model(
    "zero-shot-classification-transformer-torch",
    name_or_path="google/siglip2-base-patch16-224",  # HF model name or path
    classes=dataset.distinct("ground_truth.detections.label"),  # optional
)
dataset.apply_model(model, label_field="sigclip2_predictions")
```
All fire hydrant

All in all, as labels get larger, models out put garbage. Lots of changes in transformers in the preprocessing pipeline so needs further investigation

**Comment by danielgural** [comment]
Zero shot detection

import torch
from transformers import AutoModel, AutoProcessor
from transformers.image_utils import load_image

import fiftyone as fo
import fiftyone.zoo as foz

dataset = foz.load_zoo_dataset("quickstart") 

model = foz.load_zoo_model(
    "zero-shot-detection-transformer-torch",
    name_or_path="IDEA-Research/grounding-dino-tiny",
    classes=dataset.distinct("ground_truth.detections.label"),
)
dataset.apply_model(model, label_field="all", confidence_thresh=0.2)

All the labels are off, confidence seems fine

**Comment by danielgural** [comment]
<img width="1910" alt="Image" src="https://github.com/user-attachments/assets/06b65140-ecaa-42f0-80c3-f6d30962c425" />

**Comment by chengolivia** [comment]
I was able to reproduce the bugs you saw @danielgural . I have been investigating the issue, and here is what I've found so far.

When I ran the `load_zoo_model()` function with classes, I would get a warning:
```python
model = foz.load_zoo_model(
    "zero-shot-classification-transformer-torch",
    name_or_path="BAAI/AltCLIP",  # HF model name or path
    classes=["cat", "dog", "bird", "fish", "turtle"],  # optional
)
```
```
Classes were passed in to the FiftyOne model, but the HugginFace Transformers model configuration already has classes. Either set the classes argument to `None` to inherit the classes from the HFT model, or change the classes in the HFT model configuration to match the classes you want to use.
confidence_thresh: None
```

When I ran `load_zoo_model()` without classes, the only classes were `LABEL_0` and `LABEL_1`.
![Image](https://github.com/user-attachments/assets/51c99989-b4b4-49f9-8ec7-ab6599ef18b3)

It looks like there are default configs that these zoo models get loaded with:
https://github.com/voxel51/fiftyone/blob/develop/fiftyone/zoo/models/__init__.py#L318C5-L318C65
```python
config_dict = deepcopy(model.default_deployment_config_dict)
```

and then kwargs get merged in:
https://github.com/voxel51/fiftyone/blob/develop/fiftyone/core/models.py#L2009-L2013
```python
    if kwargs:
        if model_config_dict["type"] == etau.get_class_name(foue.ETAModel):
            _merge_config(model_config_dict["config"]["config"], kwargs)
        else:
            _merge_config(model_config_dict["config"], kwargs)
```

However, the actual config for that model, which is a FiftyOneZeroShotTransformerForImageClassification, has a Hugging Face config embedded inside it. One of the config fields is "hf_config" which itself contains the fields

```
                "id2label": {
                    "0": "LABEL_0",
                    "1": "LABEL_1"
                },
                "label2id": {
                    "LABEL_0": 0,
                    "LABEL_1": 1
                },
```

I suspect this issue is connected to this comment on the recent PR that refactored the transformers, which warned about the id2label in the HF models: https://github.com/voxel51/fiftyone/pull/5761/files#r2072212534

I plan on continuing investigation into this over the next couple days.
